hera_pspec/uvwindow.py

```diff
@@ -452,26 +350,33 @@ def __from_uvpspec__(cls, uvp, ftfile=None, ipol=0, spw=0,
         # polarisation pair
         polpair = uvputils.polpair_int2tuple(uvp.polpair_array[ipol], pol_strings=True)
 
-        # obtain spw_range from bandwidth of FTBeam object
-        # read ftbeam_obj if fed in list
-        if ftbeam_obj is None:
-            temp_ftbeam_obj = FTBeam(polpair[0], ftfile=ftfile, spw_range=None,
-                                      verbose=verbose)
-        else:
-            temp_ftbeam_obj = ftbeam_obj[0] if isinstance(ftbeam_obj, (list, tuple, np.ndarray)) else ftbeam_obj
-            assert temp_ftbeam_obj.spw_range == (0, temp_ftbeam_obj.freq_array.size),\
-                "FTBeam object fed must have no spw_range"
-            assert temp_ftbeam_obj.check(), "There is an issue with the ft_beam object."
-        
-        # read spw_range
-        bandwidth = temp_ftbeam_obj.freq_array
+        # create FTBeam objects
+        ftbeam_obj_pol = []
+        for ip, pol in enumerate(polpair):
+            if ftfile is None:
+                ftbeam_obj_pol.append(FTBeam.from_beam(beamfile='tbd',
+                                                       verbose=verbose,
+                                                       x_orientation=x_orientation))
+            else:
+                ftbeam_obj_pol.append(FTBeam.from_file('{}_{}.hdf5'.format(ftfile, pol),
+                                                       spw_range=None,
+                                                       verbose=verbose,
+                                                       x_orientation=x_orientation))
+            if pol[1] == pol[0]:
+                # do not recompute if two polarisations are identical
+                ftbeam_obj_pol.append(copy.deepcopy(ftbeam_obj_pol[0]))
+                continue
```

Comment:
I think this will add a third `ftbeam` to `ftbeam_obj_pol` in the case that `pol[1] == pol[0]`. You probably need something like 

```python
for ip, pol in enumerate(polpair):
    if ip > 0 and pol[ip] == pol[0]:
        ftbeam_obj_pol.append(ftbeam_obj_pol[0])
    else:
        if ftfile is None:
            ....

Reply:
Furthermore, not sure why you're deepcopying the ftbeam here -- it could be a waste of memory, unless you plan on users (or yourself) modifying the second beam in-place independently of the first beam. I would say this shouldn't be done, but there are cases in which it would be reasonable.

Reply:
The `deepcopy` is here because when I later run `update_spw` on the first `ftbeam_obj_pol`, without the `deepcopy`, it also updates the frequency array of the second `ftbeam_obj_pol`. Which is what we want to achieve in the end but I was worried the fact it is only a pointer might mess things up further along the way

Reply:
You're right to be cautious here (and things like this are why in general I like to have classes that are immutable, i.e. no `update_xxx`, wherever possible). I'll leave it up to your judgment, but I think that given there's just the one `update` function, and that you can dissuade users from updating any other attributes, that this can be a pointer.